setup.py

```diff
@@ -94,7 +94,7 @@ def get_git_sha() -> str:
         "parsedatetime",
         "pgsanity",
         "polyline",
-        "pyparsing>=2.4.7, <3.0.0",
+        "pyparsing>=2.4.7, <4.0.0",
```

Comment:
@villebro not sure how concerned you are about this upgrade i saw the package is only being used in one place in the codebase. Let me know what you think

Reply:
@hughhhh  I think this makes perfect sense. I actually ended up bumping that package in #17963 to unbreak `pip-compile-multi`, so if that PR happens to get merged before this one it might cause a need for a rebase.